kore/src/Kore/Step/Simplification/TermLike.hs

```diff
@@ -389,11 +397,39 @@ simplifyInternal term sideCondition = do
                 Ceil.simplify sideCondition =<< simplifyChildren ceilF
             EqualsF equalsF ->
                 Equals.simplify sideCondition =<< simplifyChildren equalsF
-            ExistsF exists ->
-                Exists.simplify sideCondition
-                    =<< simplifyChildren (refresh exists)
+            ExistsF exists -> do
+                simplifiedChildren <-
+                    simplifyChildren (refresh exists)
+                targetedResults <-
+                    Exists.simplify
+                        (targetSideCondition sideCondition)
```

Comment:
Why don't we target the children before simplification? That is, add the `Target` and `NonTarget` wrappers in `refresh. That should eliminate the need to do any re-ordering at all in the `\exists` simplifier.

Reply:
I tried doing that at first but I remember it resulted in some type issues, but maybe I did something wrong along the way and that's why it didn't work. I tried a few solutions and the current one seemed the cleanest, but I can revisit what you suggest if you think it's better.

Reply:
> I remember it resulted in some type issues

That was expected, but let's merge what you have, and come back to it.